### ISSUE TITLE
"Stick to the front page" should be contextually aware of what the front page displays

### ISSUE THREAD

**Issue by mor10** [WordPress/gutenberg#3072]
## Issue Overview
Under the Document panel, the Sticky post feature says "Stick to Front Page". This is only accurate if the site has the blog index as the front page. 

## Steps to Reproduce (for bugs)
1. Set a static page as the front page and a different static page as the blog index.
2. Create a Gutenberg post.
3. Look at the wording for the Sticky post toggle.

## Expected Behavior
When "Your homepage displays" is set to a static page and the blog index is placed on a separate static page, the wording of the Sticky post toggle should reflect this:

- When blog index is front page: "Stick to the Front Page"
- When static page is front page: "Stick to the top of blog index" or similar

## Current Behavior
"Stick to the Front Page" is displayed in all circumstances.

## Possible Solution
Make the Sticky post text contingent on the current front page settings.

**Comment by karmatosed** [comment]
I agree it should make more sense contextually. However, I think we could be a little more human in our description - 'blog index' isn't an easy term for non advanced users. What about:

'Blog posts: stick to front'

Even better, could this setting just hide when a page is set on front?

**Comment by mor10** [comment]
Now that I think about it I'm not sure exactly how the sticky post feature works. Does a sticky post only stick on top of the index if the index is the front page of the site but not if the index is a separate page? 

If so, "Blog posts: stick to front" makes sense. If the sticky post feature works regardless of where the blog index is positioned, then the feature has merit regardless of front page settings.

I'm not sure "stick to front" is descriptive enough to actually explain what's going on. It begs the question "what does 'stick' and 'front' mean?" I feel like there's some better way of explaining it - like "make first visible post" or "bring to front of post list" or something like that. The whole "stick to top/front" terminology is inherited from old school forums and has almost no contextual meaning in the modern WordPress post index world.

**Comment by aaronjorbin** [comment]
`Stick to the Front Page ` makes 100% sense if you know what `front` means, which isn't an assumption we should make. 

How about `Stick to Homepage` and `Stick to Posts page` depending on what is selected? That would make things a bit more consistent with the options page (screenshot below)  

<img width="574" alt="screen shot 2017-10-24 at 4 06 20 pm" src="https://user-images.githubusercontent.com/622599/31965414-59b09eb2-b8d5-11e7-8baa-1876b9ecd8ff.png">

**Comment by mor10** [comment]
That sounds like a clean solution to me.

**Comment by WPprodigy** [comment]
Would it be out of scope to expose the `show_on_front` WP setting to the REST API's [settings endpoint](https://developer.wordpress.org/rest-api/reference/settings/) via `register_rest_field` in this plugin?

Not 100% sure what's best practice for a feature plugin like this. As far as I can tell Gutenberg hasn't needed to do this yet as all the data it needs has been available from the REST API.

**Comment by haicu** [comment]
I agree with Clorith. In the gutenberg editor screen one of the first things I encounter for a post is the option to "stick to the front page". This is confusing wording. What actually happens when you make a post sticky is that it is put on top of the post list. So I would suggest that the wording reflects that behaviour.

So what about "stick this post to the top of the post list". Or maybe a shortened version of this?

If theme authors bypass this sticky feature, they must be able to turn it off. Is there an add_theme_support_for_sticky_post function or something like that to turn it off?

**Comment by melchoyce** [comment]
How about "Always show this post at the top of my blog?"

**Comment by haicu** [comment]
Better yes!

**Comment by Clorith** [comment]
I'm not 100% sold, not unlikely scenario: What blog? I use WordPress as a CMS, I don't blog, I post news articles about my company.

I don't have any better suggestions, unfortunately, but it's something I'd love if we could find soem kind of neutral terminology for if at all possible.

**Comment by haicu** [comment]
How about a more generic phrase like "Always show this post at the top of my post list?"

**Comment by mtias** [comment]
There doesn't seem to be a consensus on the best way to handle this yet. This will be a center point for the next phases, so it'll have to be approached more holistically then.

If there's a simple copy suggestion that would improve things, let's consider it.

**Comment by WPprodigy** [comment]
Hmm, it only shows on posts though, not pages.

**Comment by adleviton** [comment]
True. But not all websites work in the context of being able to stick posts to the front page. So from the client's or end user's perspective, it's just one more thing that can cause confusion and UI clutter.

**Comment by pkimbrell38** [comment]
How about if you don't want this option to show up at all in the ep editor? For example, some wp sites are multi-author sites and I have been trying to figure out how to make this hidden completely in the wp editor, "Stick to the top of the blog." I have been unable to find a solution anywhere.

**Comment by noisysocks** [comment]
> How about a more generic phrase like "Always show this post at the top of my post list?"

I like this copy. Posts can be displayed in lost of places now that we e.g. have a Query block.

> How about if you don't want this option to show up at all in the ep editor? For example, some wp sites are multi-author sites and I have been trying to figure out how to make this hidden completely in the wp editor, "Stick to the top of the blog." I have been unable to find a solution anywhere.

I don't believe there is currently a way to disable this. WordPress Core would have to add this functionality. Please open a ticket in https://core.trac.wordpress.org/newticket.

**Comment by Mamaduka** [comment]
The wording was updated to `Stick to the top of the blog` in #13120.

**Comment by Mamaduka** [comment]
@mor10, can we close this issue?


## Screenshot
![CleanShot 2023-06-13 at 11 18 57](https://github.com/WordPress/gutenberg/assets/240569/d80871f4-6cea-4358-a4c6-ad37d9a3d25f)

**Comment by mor10** [comment]
Looks good to me
